src/App.js

```diff
@@ -1,84 +1,86 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
+
+// import weatherData from "./WeatherData";
+import Weather from "./components/Weather";
+import Loading from "./components/Loading";
+import Forecast from "./components/Forecast";
+import { filterForecast } from "./utils/helpers";
+import {cities} from "./utils/cities"
 import "./App.css";
 
 const App = () => {
+  
+  const API_KEY = process.env.REACT_APP_MY_API_ID
+  const API_BASE_CURRENT = 'https://api.openweathermap.org/data/2.5/weather?units=metric&APPID=';
+  const API_BASE_FORECAST = 'https://api.openweathermap.org/data/2.5/forecast?units=metric&APPID=';
+ 
+  // const [weather, setWeather] = useState(weatherData);
+  const [weather, setWeather] =  useState(null);
+  const [city, setCity] =  useState("Sydney");
+  const [forecast, setForecast] = useState(null);
+  
+  const fetchWeather = () => {
+    fetch(
+      API_BASE_CURRENT + API_KEY + '&q=' + city
+    )
+      .then((response) => response.json())
+      .then((json) => {
+        setWeather(json);
+        });
+  };
+
+  const fetchForecast = () => {
+    fetch (
+      API_BASE_FORECAST + API_KEY + '&q=' + city
+    )
+    .then((response) => response.json())
+    .then((json) => {
+      setForecast(filterForecast(json.list));
+    });
+  }
+
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  useEffect(fetchWeather, [city]);
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  useEffect(fetchForecast, [city]);
+  
+  const handleButtonClick = (event) => {
+    setCity(event.target.outerText);
```

Comment:
nenene, ani náhodou, grafickým obsahem neřiď data. (je to kdyžtak vždy opačně - data řídí grafický obsah)
`<button className="button" onClick={() => handleButtonClick("Prague")}>Prague</button>`
protože pak někdo přidá k Prague mezeru, ikonu, obalí to <i> a jsi nahraná.